src/dict.h

```diff
@@ -108,6 +108,19 @@ typedef struct {
     dictDefragAllocFunction *defragVal;   /* Defrag-realloc values (optional) */
 } dictDefragFunctions;
 
+/* Iterator for randomized full-coverage dictionary traversal.
+ *
+ * Visits every entry in the dictionary exactly once, starting from a
+ * random offset and wrapping around. Useful for fairness in periodic tasks.
+ */
+typedef struct dictRandomIterator {
+    dict *d;
+    dictIterator *iter;  /* For phase 0: from skip to end */
```

Comment:
The comment is incorrect here. We reuse the same iter in the second phase as well.

```
    dictIterator *iter;  /* iterator to access the dict entries */
```